javaslang/src/main/java/javaslang/collection/Multimaps.java

```diff
@@ -0,0 +1,171 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.collection.Multimap.ContainerType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * Implementations of common {@code Map} functions (not intended to be public).
+ *
+ * @author Ruslan Sennov, Daniel Dietrich
+ * @since 2.1.0
+ */
+final class Multimaps {
+
+    static <K, V, K2, V2, M extends Multimap<K2, V2>> M bimap(
+            Multimap<K, V> multimap,
+            OfEntries<K2, V2, M> ofEntries,
+            Function<? super K, ? extends K2> keyMapper,
+            Function<? super V, ? extends V2> valueMapper) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
```

Comment:
I didn't do it on purpose. There are parameters that are passed internally by the Multimap impls and there are user-parameters that are routed through. We only test the latter because we are sure that the former are correct.